src/pkg/sypgp/sypgp.go

```diff
@@ -0,0 +1,424 @@
+		}
+	}
+
+	// Revocation Signatures
+	for _, s := range entity.Revocations {
+		putSigInfo(s)
+		fmt.Printf("--------- --------- ---------\n")
+	}
+	fmt.Println("")
+
+	// Subkeys Signatures
+	for _, sub := range entity.Subkeys {
+		putSigInfo(sub.Sig)
+		fmt.Printf("--------- --------- [%s]\n", entity.PrimaryKey.KeyIdShortString())
+	}
+
+	fmt.Println("<=++++++++++++++++++++++++++++++++++++++++++++++++++")
+
+	return nil
+}
+
+// DirPath returns a string describing the path to the sypgp home folder
+func DirPath() string {
+	return filepath.Join(os.Getenv("HOME"), ".sypgp")
+}
+
+// SecretPath returns a string describing the path to the private keys store
+func SecretPath() string {
+	return filepath.Join(DirPath(), "pgp-secret")
+}
+
+// PublicPath returns a string describing the path to the public keys store
+func PublicPath() string {
+	return filepath.Join(DirPath(), "pgp-public")
+}
+
+// PathsCheck creates the sypgp home folder, secret and public keyring files
+func PathsCheck() error {
+	if err := os.MkdirAll(DirPath(), 0700); err != nil {
+		log.Println("could not create singularity PGP directory")
+		return err
+	}
+
+	fs, err := os.OpenFile(SecretPath(), os.O_RDWR|os.O_CREATE, 0600)
+	if err != nil {
+		log.Println("Could not create private keyring file: ", err)
+		return err
+	}
+	fs.Close()
+
+	fp, err := os.OpenFile(PublicPath(), os.O_RDWR|os.O_CREATE, 0600)
+	if err != nil {
+		log.Println("Could not create public keyring file: ", err)
+		return err
+	}
+	fp.Close()
+
+	return nil
+}
+
+// LoadPrivKeyring loads the private keys from local store into an EntityList
+func LoadPrivKeyring() (openpgp.EntityList, error) {
+	if err := PathsCheck(); err != nil {
+		return nil, err
+	}
+
+	f, err := os.Open(SecretPath())
+	if err != nil {
+		log.Println("Error trying to open secret keyring file: ", err)
+		return nil, err
+	}
+	defer f.Close()
+
+	el, err := openpgp.ReadKeyRing(f)
+	if err != nil {
+		log.Println("Error while trying to read secret key ring: ", err)
+		return nil, err
+	}
+
+	return el, nil
+}
+
+// LoadPubKeyring loads the public keys from local store into an EntityList
+func LoadPubKeyring() (openpgp.EntityList, error) {
+	if err := PathsCheck(); err != nil {
+		return nil, err
+	}
+
+	f, err := os.Open(PublicPath())
+	if err != nil {
+		log.Println("Error trying to open public keyring file: ", err)
+		return nil, err
+	}
+	defer f.Close()
+
+	el, err := openpgp.ReadKeyRing(f)
+	if err != nil {
+		log.Println("Error while trying to read public key ring: ", err)
+		return nil, err
+	}
+
+	return el, nil
+}
+
+func printEntity(index int, e *openpgp.Entity) {
+	for _, v := range e.Identities {
+		fmt.Printf("%v) U: %v %v %v\n", index, v.UserId.Name, v.UserId.Comment, v.UserId.Email)
+	}
+	fmt.Printf("   C: %v\n", e.PrimaryKey.CreationTime)
+	fmt.Printf("   F: %0X\n", e.PrimaryKey.Fingerprint)
+	bits, _ := e.PrimaryKey.BitLength()
+	fmt.Printf("   L: %v\n", bits)
+}
+
+func printPubKeyring() (err error) {
```

Comment:
Since you've got a named return value (and this is a short `func`), you could do naked returns in this function.